core/txpool/bundlepool/bundlepool.go

```diff
@@ -73,27 +73,54 @@ type BundlePool struct {
 	config Config
 
 	bundles    map[common.Hash]*types.Bundle
-	bundleHeap BundleHeap
+	bundleHeap BundleHeap // least price heap
 	mu         sync.RWMutex
 
 	slots uint64 // Number of slots currently allocated
 
-	simulator BundleSimulator
+	// bundleMetrics for mev data analyst
+	bundleMetrics   map[int64][]*types.Bundle // receivedBlockNumber -> bundle
+	bundleMetricsMu sync.RWMutex
+
+	simulator  BundleSimulator
+	blockchain BlockChain
 }
 
-func New(config Config) *BundlePool {
+func New(config Config, chain BlockChain) *BundlePool {
 	// Sanitize the input to ensure no vulnerable gas prices are set
 	config = (&config).sanitize()
 
 	pool := &BundlePool{
 		config:     config,
 		bundles:    make(map[common.Hash]*types.Bundle),
 		bundleHeap: make(BundleHeap, 0),
+		blockchain: chain,
 	}
 
+	go pool.clearLoop()
+
 	return pool
 }
 
+func (p *BundlePool) clearLoop() {
+	ticker := time.NewTicker(5 * time.Minute)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		least := p.blockchain.CurrentBlock().Number.Int64()
+		for number := range p.bundleMetrics {
+			if number < least {
+				least = number
+			}
+		}
+
+		for len(p.bundleMetrics) > types.MaxBundleAliveBlock {
```

Comment:
it is possible that we dont have bundle for a long time, so that the bundle long time ago still wont be prune, guess this is not expected.

Always prune the block that is less than `current height - MaxBundleAliveBlock`

Reply:
fixed